runtime/README.md

```diff
@@ -106,56 +108,60 @@ The contexts contain the internal structures necessary to resume operations to s
 
 These contexts are intended to be opaque to the user, and SHALL be encrypted and authenticated if they contain sensitive internal data.
 
-### Data Storage
-
-The CM system has its own dedicated storage in DCCM.
-
-This storage is protected with authenticated encryption using internal keys that are randomly generated at power on and stored in the key vault.
-
-This storage is exposed to users through opaque 256-bit handles, called cryptographic mailbox IDs (CMIDs). These are mapped internally into DCCM storage blocks.
-
-Keys can be imported using `CM_IMPORT` and deleted via `CM_DELETE`.
-
-The entire contents can be cleared with `CM_CLEAR`.
+### Keys
 
-The status of the data storage can be queried with `CM_STATUS`.
+Cryptographic Mailbox Key (CMKs) are used to store keys. Certain commands generate and return a new CMK. Most commands that use CMKs will also return a new CMK, as it is necessary to track CMKs so that they are not used beyond any relevant limits for their key type.
 
-### Storage Design
+They are returned from commands that generate keys and must be passed back to Caliptra to be used. These keys are encrypted and opaque to the mailbox caller.
 
-Data is stored in 16-byte blocks.
+Internally, the unecrypted CMKs have the following structure:
 
-Each block's status is tracked in a bitfield, 1 bit per block, with a value of 1 indicating that block is in use.
+| **Name**      | **Bits** | **Description**                             |
+| ------------- | -------- | ------------------------------------------- |
+| version       | 32       | CMK version. Currently always 1.            |
+| key usage     | 32       | represents which kind of key this is        |
```

Comment:
I think some fields here like version or key usage would be useful if they were in cleartext, for debugging. ("Why didn't my command work? Am I passing the wrong kind of CMK to the command?")

In fact, we could potentially move everything but the key material to the outer structure, and make it all AAD.

Reply:
For here it is relatively safe to keep some of this in plaintext (and make it AAD, certainly, to prevent modification). But, for AES at least, some of the fields are sensitive. It feels weird to me to have different code paths to encrypt the contexts for different commands -- I'd rather they we were all encrypted.